### DISCUSSION TITLE
3.5 抽象语言的解释器

### DISCUSSION THREAD

**Discussion by giscus** [csfive/composing-programs-zh#119] [Announcements]
# 3.5 抽象语言的解释器

SICP Python 中文翻译

https://composingprograms.netlify.app/3/5

**Comment by 635237669** [comment]
这里写的太概括导致不太对。一个通用的用A语言解释B语言的解释器。需要一个元循环的A解释器（sicp写的）与一个用A写的B解释器（61A项目写的），然后用前者来模拟后者，进而模拟任何用B写的程序。

**Comment by howwilli** [comment]
大开眼界！真的太酷了！！！！！！！！！
    we can regard the Scheme interpreter as a very special machine that takes as input a description of a machine
    Given this input, the interpreter configures itself to emulate the machine described
    From this perspective, our Scheme interpreter is seen to be a universal machine. It mimics other machines when these are described as Scheme programs.
